### PR TITLE
Add method to set plugin variables

### PR DESCRIPTION
**Pull request by nicohenao**
An extra method is added to the API that allows to set a variable for a specified plugin in `config.xml` file.

### CODE REVIEW

File: .jshintrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "esversion":6 }
```
Comment (SamVerschueren) [inline .jshintrc:1]: I'm not using jshint, so remove this file.

File: index.js [hunk 2]
```diff
@@ -29,7 +27,7 @@ module.exports = (function () {
 				contents = contents.substring(contents.indexOf('<'));
 			}
 
-			var doc = new et.ElementTree(et.XML(contents));		// eslint-disable-line babel/new-cap
+			var doc = new et.ElementTree(et.XML(contents)); // eslint-disable-line babel/new-cap
```
Comment (SamVerschueren) [inline index.js:30]: Undo change

File: index.js [hunk 3]
```diff
@@ -1,5 +1,3 @@
-'use strict';
```
Comment (SamVerschueren) [inline index.js:1]: Do not remove

File: index.js [hunk 4]
```diff
@@ -342,7 +378,7 @@ module.exports = (function () {
 	 */
 	Config.prototype.addRawXML = function (raw) {
 		// Parse the raw XML
-		var xml = et.XML(raw);		// eslint-disable-line babel/new-cap
+		var xml = et.XML(raw); // eslint-disable-line babel/new-cap
```
Comment (SamVerschueren) [inline index.js:381]: Undo

File: index.js [hunk 5]
```diff
@@ -354,14 +390,18 @@ module.exports = (function () {
 	 * @returns {Promise}			A promise that resolves when the file is written.
 	 */
 	Config.prototype.write = function () {
-		return pify(fs.writeFile, Promise)(this._file, this._doc.write({indent: 4}), 'utf8');
+		return pify(fs.writeFile, Promise)(this._file, this._doc.write({
```
Comment (SamVerschueren) [inline index.js:393]: Undo

File: index.js [hunk 6]
```diff
@@ -354,14 +390,18 @@ module.exports = (function () {
...
+			indent: 4
+		}), 'utf8');
 	};
 
 	/**
 	 * The same as `write` but sync.
 	 */
 	Config.prototype.writeSync = function () {
-		fs.writeFileSync(this._file, this._doc.write({indent: 4}), 'utf-8');
+		fs.writeFileSync(this._file, this._doc.write({
```
Comment (SamVerschueren) [inline index.js:402]: Undo

File: test/set-plugin-variable-value.js [hunk 7]
```diff
@@ -0,0 +1,24 @@
+import test from 'ava';
```
Comment (SamVerschueren) [inline test/set-plugin-variable-value.js:1]: Can you add an extra test which sets multiple plugin variables?

File: README.md [hunk 8]
```diff
@@ -75,6 +75,31 @@ Type: `object`
 
 The element attributes
 
+### #setPluginVariable(pluginName, variableName, variableValue)
```
Comment (SamVerschueren) [inline README.md:78]: I believe it would make more sense to use `addPluginVariable` instead. Because it's possible to call it multiple times.

**Review by SamVerschueren — CHANGES_REQUESTED** [review]
Thanks for the PR! Added some comments. Would it also be possible to remove the `package-lock.json` file?

**Comment by morellan** [comment]
Any news on this @nihenaoar ?, I'd love to have this feature merged on the main repo

**Comment by nicohenao** [comment]
@morellan I've updated the PR, it's been moved to #23 since this one comes from a fork that's not mine.

**Pull request closed without merging**